### ISSUE TITLE
Refresh Path button not working on multiplayer server

### ISSUE THREAD

**Issue by Kiritow** [Minecraft-Transit-Railway/Minecraft-Transit-Railway#352]
**Expected Behaviour**
After clicking the `refresh path` button, it should turn gray and turn back to normal state in less than 1 second.

**Actual Behaviour**
After clicking the `refresh path` button, it turned gray and couldn't turned back into normal state until next login.

**Reproduction Steps**
Steps to reproduce the behaviour:
1. Open `railway dashboard` and switched to `Depots` tab.
2. Make changes to routes, and click `Refresh Path`

**Crash Report**
No crash report. Server side logs:
```
Starting path generation for Depot
Finished path generation for Depot
```
Client side error logs:
```
[15:37:50] [Render thread/ERROR]: Error executing task on Client
java.lang.IndexOutOfBoundsException: readerIndex(23) + length(1) exceeds writerIndex(23): PooledUnsafeDirectByteBuf(ridx: 23, widx: 23, cap: 23)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1442) ~[netty-all-4.1.68.Final.jar%2326!/:4.1.68.Final]
	at io.netty.buffer.AbstractByteBuf.readByte(AbstractByteBuf.java:730) ~[netty-all-4.1.68.Final.jar%2326!/:4.1.68.Final]
	at net.minecraft.network.FriendlyByteBuf.readByte(FriendlyByteBuf.java:909) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.network.FriendlyByteBuf.m_130242_(FriendlyByteBuf.java:344) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.network.FriendlyByteBuf.m_130136_(FriendlyByteBuf.java:486) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.network.FriendlyByteBuf.m_130277_(FriendlyByteBuf.java:482) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at mtr.packet.PacketTrainDataGuiClient.createRailS2C(PacketTrainDataGuiClient.java:120) ~[MTR-forge-1.18.2-3.0.0.jar%2372!/:?]
	at mtr.MTRClient.lambda$init$22(MTRClient.java:288) ~[MTR-forge-1.18.2-3.0.0.jar%2372!/:?]
	at mtr.forge.RegistryClientImpl.lambda$registerNetworkReceiver$0(RegistryClientImpl.java:47) ~[MTR-forge-1.18.2-3.0.0.jar%2372!/:?]
	at dev.architectury.networking.forge.NetworkManagerImpl.lambda$createPacketHandler$6(NetworkManagerImpl.java:150) ~[architectury-4.3.53.jar%2354!/:?]
	at dev.architectury.networking.transformers.PacketTransformer$1.inbound(PacketTransformer.java:47) ~[architectury-4.3.53.jar%2354!/:?]
	at dev.architectury.networking.forge.NetworkManagerImpl.lambda$createPacketHandler$7(NetworkManagerImpl.java:145) ~[architectury-4.3.53.jar%2354!/:?]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.network.NetworkInstance.dispatch(NetworkInstance.java:68) ~[forge-1.18.2-40.1.0-universal.jar%2384!/:?]
	at net.minecraftforge.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:75) ~[forge-1.18.2-40.1.0-universal.jar%2384!/:?]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at net.minecraftforge.network.NetworkHooks.onCustomPayload(NetworkHooks.java:75) ~[forge-1.18.2-40.1.0-universal.jar%2384!/:?]
	at net.minecraft.client.multiplayer.ClientPacketListener.m_7413_(ClientPacketListener.java:1824) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket.m_5797_(ClientboundCustomPayloadPacket.java:57) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket.m_5797_(ClientboundCustomPayloadPacket.java:7) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.network.protocol.PacketUtils.m_131356_(PacketUtils.java:22) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:157) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:131) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:116) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1013) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:663) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at net.minecraft.client.main.Main.main(Main.java:205) ~[client-1.18.2-20220404.173914-srg.jar%2380!/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:31) ~[fmlloader-1.18.2-40.1.0.jar%2316!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:149) [bootstraplauncher-1.0.0.jar:?]
```

**Screenshots**
![image](https://user-images.githubusercontent.com/13619158/166419143-4ac8bb5c-d9b3-4eef-8d41-d63fa92f312f.png)

**Environment**

- Minecraft Version: 1.18.2
- Forge Version: 40.1.0
- `Authlib-injector` (1.1.43) is installed to implement custom login.
- Installed Mods (Server Side):
```
appleskin-forge-mc1.18-2.4.0.jar
appliedenergistics2-11.0.0-alpha.6.jar
architectury-4.3.53.jar
AutoRegLib-1.7-53.jar
badpackets-forge-0.1.2.jar
balm-3.0.3+0.jar
Botania-1.18.2-431.jar
craftingtweaks-forge-1.18.1-14.0.2.jar
curios-forge-1.18.2-5.0.7.0.jar
excavation-1.18-1.7.2.jar
FastLeafDecay-28.jar
jei-1.18.2-9.7.0.194.jar
journeymap-1.18.2-5.8.2-forge.jar
JustEnoughResources-1.18.1-0.13.1.141.jar
mcjtylib-1.18-6.0.13.jar
Mekanism-1.18.2-10.2.0.459.jar
MekanismAdditions-1.18.2-10.2.0.459.jar
MekanismGenerators-1.18.2-10.2.0.459.jar
MekanismTools-1.18.2-10.2.0.459.jar
MouseTweaks-forge-mc1.18-2.21.jar
MTR-forge-1.18.2-3.0.0.jar
Neat+1.8-30.jar
Patchouli-1.18.2-67.jar
Quark-3.2-353.jar
rftoolsbase-1.18-3.0.7.jar
StorageDrawers-1.18.2-10.2.0.jar
wthit-forge-4.9.1.jar
xnet-1.18-4.0.5.jar
```

**Additional Information**
A server restart might fix the problem, but after sometime it comes back again.

**Comment by Kiritow** [comment]
Found a worked but strange solution here: https://www.minecraftforum.net/forums/support/java-edition-support/3061493-i-cannot-connect-to-a-modded-server
After deleteing my player dat file from server and restart it, everything went back normal again!

**Comment by Kiritow** [comment]
Sorry after a few hours the problem comes back again... and other players who had used rail plan board also met this problem.

**Comment by Kiritow** [comment]
![image](https://user-images.githubusercontent.com/13619158/166701049-6bc87d6d-6386-497c-ba91-cdfb5511cb40.png)
After adding some debug code and replacing the mod with a recompiled one, it seems that this problem is caused by incomplete packets.

**Comment by jonathanqwq** [comment]
yes just i've met this problem minutes ago

but the process was finished after 2 minutes 

strange problem lol

**Comment by jonafanho** [comment]
Are you using a modified version of the jar, either on server or client?

**Comment by jonafanho** [comment]
This should be fixed in 4.0.0. Please reopen this bug if you still encounter this issue with 4.0.0, thanks!
